bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/WeatherApi.java

```diff
@@ -73,6 +80,31 @@ public NAMain getStationData(String deviceId) throws NetatmoException {
         throw new NetatmoException("Unexpected answer searching device '%s' : not found.", deviceId);
     }
 
+    /**
+     * Returns data from a Weather Station owned by the user
+     *
+     * This method must be preferred to getStationData when you know that the device is a station owned by the user
+     * (because it avoids requesting additional data for favorite/guest stations).
+     *
+     * @param deviceId Id of the device you want to retrieve information
+     * @return NAMain
+     * @throws NetatmoException If fail to call the API, e.g. server error or deserializing
+     */
+    public NAMain getOwnedStationData(String deviceId) throws NetatmoException {
+        ListBodyResponse<NAMain> answer = getStationsData(deviceId, false).getBody();
+        if (answer != null) {
+            NAMain station = answer.getElement(deviceId);
+            if (station != null) {
+                return station;
+            }
+        }
+        throw new NetatmoException("Unexpected answer searching device '%s' : not found.", deviceId);
+    }
+
+    public @Nullable ListBodyResponse<NAMain> getFavoriteAndGuestStationsData() throws NetatmoException {
+        return getStationsData(null, true).getBody();
```

Comment:
I would suggest to drop `@Nullable` and instead throw exception if null like the other method variants.

Reply:
I am not sure. If you have no weather station s attached to your account, maybe the result will be null ? I don't know if the API will return null or an empty list. And I can't test it.

Reply:
But Gael was testing if the body was not null in `startScan`, so I assume it could be null.

Reply:
This is of course different for getStationData because it is called for a previously identified weather station.

So I would suggest to keep the code as it is now.

Reply:
You are right that `startScan` doesn't currently consider this as an error. However, it needs to handle the case. So I would then suggest to return an empty list from `getFavoriteAndGuestStationsData` when null is returned from `getStationsData`. This way the caller doesn't have to deal with this, and it will be clear that this is not considered an error. Currently caller has three possible outcomes: null, a list or an exception thrown, which can easily be reduced to two.

Reply:
Done. I hope this is the last change now ;)

Reply:
It sure is. :-) Thanks for applying all the changes for my suggestions. Not only is the problem now fixed, but the confusion that led to the problem is also addressed by the refactoring.